FirebaseCore/Internal/Sources/HeartbeatLogging/RingBuffer.swift

```diff
@@ -62,7 +63,7 @@ struct RingBuffer<Element>: Sequence {
     // Decrement index, wrapping around to the back if needed.
     tailIndex -= 1
     if tailIndex < 0 {
-      tailIndex = circularQueue.count - 1
+      tailIndex = circularQueue.endIndex - 1
```

Comment:
Based on https://stackoverflow.com/questions/24853121/what-is-the-correct-behaviour-of-endindex-of-array-in-swift should not be any different

Reply:
Correct